sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_impl.hpp

```diff
@@ -95,8 +100,9 @@ int main() {
     init_and_multiply<bfloat16, float, 2, 1, 500,
                       16>(); // 500 is not correct size
   } catch (const sycl::exception &e) {
-    if (e.code() == errc::kernel_not_supported)
+    if (e.code() == errc::invalid)
       return 0;
+    throw;
```

Comment:
```suggestion
```

Reply:
The throw is required. Because if we do not catch the right error, we will just silently fail the test with `return 1;`. The throw allows us to fail the test and have some output that is useful for debugging an understanding what the unexpected exception is.

Reply:
sounds good!